clients/src/main/java/org/apache/kafka/clients/consumer/internals/AbstractStickyAssignor.java

```diff
@@ -162,6 +162,17 @@ private boolean allSubscriptionsEqual(Set<String> allTopics,
         return isAllSubscriptionsEqual;
     }
 
+    // visible for testing
+    MemberData memberDataFromSubscription(Subscription subscription) {
+        if (!subscription.ownedPartitions().isEmpty() && subscription.generationId().isPresent()) {
```

Comment:
> For instance, this could happen if the group has more members than partitions.

In that case, the `subscription.generationId()` should still contain valid number. Is that right?

Reply:
I know what you mean now. Yes, for `subscription v3` check, we should only check generationId. Will update it and add a test for it.